src/libs/sqliteHandler.go

```diff
@@ -0,0 +1,745 @@
+package libs
+
+import (
+	"database/sql"
+	"encoding/json"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/accuknox/auto-policy-discovery/src/types"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const WorkloadProcessFileSetSQLite_TableName = "workload_process_fileset"
+const TableNetworkPolicySQLite_TableName = "network_policy"
+const TableSystemPolicySQLite_TableName = "system_policy"
+
+// ================ //
+// == Connection == //
+// ================ //
+
+var MockSqlite sqlmock.Sqlmock = nil
+var MockDBSQLite *sql.DB = nil
+
+func NewMockSQLite() (*sql.DB, sqlmock.Sqlmock) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		log.Error().Msgf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+
+	MockSqlite = mock
+	MockDBSQLite = db
+
+	return db, mock
+}
+
+func waitForDBSQLite(db *sql.DB) {
+	for {
+		err := db.Ping()
+		if err != nil {
+			time.Sleep(time.Second * 1)
+			log.Error().Msgf("db.Ping() failed. Will retry. err=%s", err.Error())
+		} else {
+			break
+		}
+	}
+}
+
+func connectSQLite(cfg types.ConfigDB) (db *sql.DB) {
+	if MockDB != nil {
+		return MockDB
+	}
+
+	dbconn := cfg.DBUser + ":" + cfg.DBPass + "@tcp(" + cfg.DBHost + ":" + cfg.DBPort + ")/" + cfg.DBName
+	db, err := sql.Open(cfg.DBDriver, cfg.SQLiteDBPath)
+	for err != nil {
+		log.Error().Msgf("sqlite driver:%s, user:%s, host:%s, port:%s, dbname:%s conn-error:%s",
+			cfg.DBDriver, cfg.DBUser, cfg.DBHost, cfg.DBPort, cfg.DBName, err.Error())
+		time.Sleep(time.Second * 1)
+		db, err = sql.Open(cfg.DBDriver, dbconn)
+	}
+
+	db.SetMaxIdleConns(0)
+
+	waitForDBSQLite(db)
+
+	return db
+}
+
+// ==================== //
+// == Network Policy == //
+// ==================== //
+
+func GetNetworkPoliciesFromSQLite(cfg types.ConfigDB, cluster, namespace, status string) ([]types.KnoxNetworkPolicy, error) {
+	db := connectSQLite(cfg)
+	defer db.Close()
+
+	policies := []types.KnoxNetworkPolicy{}
+	var results *sql.Rows
+	var err error
+
+	query := "SELECT apiVersion,kind,flow_ids,name,cluster_name,namespace,type,rule,status,outdated,spec,generatedTime FROM " + TableNetworkPolicySQLite_TableName
```

Comment:
cc @nyrahul Should we be concerned of SQL injection and create prepared statements objects (using sqlite3_prepare()) for the queries?

Reply:
This particular statement won't result in SQLI ... go-sec automatically identifies scenarios where SQLI could be used ... Typically SQLI is used where the query is constructed from non-constant variables. In this case, the query is prepared from constant strings + constant var only and hence no risk.